benchmarks-api/src/main/java/io/scalecube/benchmarks/BenchmarksState.java

```diff
@@ -268,12 +268,25 @@ public final void runForAsync(Function<S, Function<Long, Publisher<?>>> func) {
       self.start();
 
       Function<Long, Publisher<?>> unitOfWork = func.apply(self);
-
-      Flux<Long> fromStream =
-          Flux.fromStream(LongStream.range(0, settings.numOfIterations()).boxed());
-
-      Flux.merge(fromStream.publishOn(scheduler()).map(unitOfWork))
-          .take(settings.executionTaskDuration())
+      int threads = settings.numberThreads();
+      long countPerThread = settings.numOfIterations() / threads;
+
+      Function<Integer, Mono<?>> scenarioPerThread =
+          i ->
+              Mono.fromRunnable(
+                  () -> {
+                    long start = i * countPerThread;
+                    Flux.fromStream(LongStream.range(start, start + countPerThread).boxed())
+                        .flatMap(unitOfWork::apply, 64, Integer.MAX_VALUE)
```

Comment:
Cool finding! 
But why not just `runtime.available_processoors` ?

Reply:
By default, this value is 256, and how I understand this value specifies how many publishers can be processed simultaneously in `flatMap` operator (other words, `flatMap` subscribe to these publishers)

Reply:
Also, I think we should add an opportunity to specify it as a benchmark settings param.